coalahtml/_coalahtml/app/controllers/main.js

```diff
@@ -2,51 +2,50 @@
 
 angular.module('coalaHtmlApp')
   .controller('MainCtrl', function ($scope, $http) {
-    var parseCoalaProject = function(projectDir, coalaJSON) {
-      coalaJSON = coalaJSON || projectDir + '/coala.json';
-      $scope.data = [];
-      $http.get(coalaJSON).success(function(response){
-        var knownFiles = {};
-        $scope.content = {};
-
-        var parseResult = function(el){
-          el.affected_code.forEach(function(sourceRange){
-              if (!(sourceRange.file in knownFiles)){
-                  knownFiles[sourceRange.file] = [];
-
-                  $http.get(projectDir + "/" + sourceRange.file)
-                    .success(function(content) {
-                      $scope.content[sourceRange.file] = content.split("\n");
-                    });
-              }
-              knownFiles[sourceRange.file].push({
-                "start":    sourceRange.start.line,
-                "end":      sourceRange.end.line,
-                "diffs":    el.diffs,
-                "message":  el.message,
-                "origin":   el.origin,
-                "severity": el.severity
+      var parseCoalaProject = function(projectDir, coalaJSON) {
+          $http.get(projectDir + '/file_data.json')
+            .success(function(file_data) {
+                $scope.file_dict = {};
+                Object.keys(file_data).forEach(function(file_name) {
+                    if (!(file_name in $scope.file_dict)){
+                      $scope.file_dict[file_name]=file_data[file_name];
+                    }
+                });
+              coalaJSON = coalaJSON || projectDir + '/coala.json';
+              $scope.data = [];
+              $http.get(coalaJSON).success(function(response){
+                  var knownFiles = {};
+                  var parseResult = function(el){
+                      el.affected_code.forEach(function(sourceRange){
+                          if (!(sourceRange.file in knownFiles)){
+                              knownFiles[sourceRange.file] = [];
+                          }
+                          knownFiles[sourceRange.file].push({
+                            "start":    sourceRange.start.line,
+                            "end":      sourceRange.end.line,
+                            "diffs":    el.diffs,
+                            "message":  el.message,
+                            "origin":   el.origin,
+                            "severity": el.severity
+                          });
+                      });
+                };
+                for (var errors in response.results) {
+                  if (response.results[errors][0]){ // If error exists
```

Comment:
So `response.results` contains errors. And you call `parseResult` on them eventually. I think there my be something wrong with naming. Also, maybe you can rewrite this in the way that the comment wouldn't be needed?